### ISSUE TITLE
Access Violation in AspNetCoreModule if dotnet exits with error

### ISSUE THREAD

**Issue by moozzyk** [aspnet/IISIntegration#137]
If the application is misconfigured so that it is not able to start w3wp may crash with AV. 

If I run my application from command line like this dotnet exits with an error code:
```
PS C:\temp\HelloMvc> dotnet ./HelloMvcaa.dll
Could not load host policy library from [C:\temp\HelloMvc]
PS C:\temp\HelloMvc> $LastExitCode
-2147450749
```
If I configure IIS to run the same way:
```
<aspNetCore processPath="dotnet.exe" arguments=".\HelloMvc.dllaa" stdoutLogEnabled="true" stdoutLogFile=".\logs\stdout" forwardWindowsAuthToken="false" startupTimeLimit="3600" />
```

I get this:

![image](https://cloud.githubusercontent.com/assets/1438884/14576577/1a47a77e-031f-11e6-8021-e4050a03aa12.png)

```
Faulting application name: w3wp.exe, version: 10.0.10586.0, time stamp: 0x5632d85f
Faulting module name: aspnetcore.dll, version: 7.1.1962.0, time stamp: 0x56f47164
Exception code: 0xc0000005
Fault offset: 0x00000000000109c5
Faulting process id: 0xef00
Faulting application start time: 0x01d197657a7bfd5a
Faulting application path: c:\windows\system32\inetsrv\w3wp.exe
Faulting module path: C:\WINDOWS\system32\inetsrv\aspnetcore.dll
Report Id: 0e92130a-1339-438f-b72d-36f9fc0431ee
Faulting package full name: 
Faulting package-relative application ID: 
```

```
Fault bucket , type 0
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: w3wp.exe
P2: 10.0.10586.0
P3: 5632d85f
P4: aspnetcore.dll
P5: 7.1.1962.0
P6: 56f47164
P7: c0000005
P8: 00000000000109c5
P9: 
P10: 
```

**Comment by PinpointTownes** [comment]
FYI, I'm seeing a similar error when stopping an app pool (hosting a working vNext app).
I assume this is caused by the same root cause.

**Comment by GuardRex** [comment]
@PinpointTownes That's interesting ... I thought stopping the AppPool would result in a 503 with no Application log entry.

**Comment by PinpointTownes** [comment]
@GuardRex a 503 response is indeed what I'd get if I tried to reach the website *after* shutting down the app pool. The exception (and the debug window) appears just after stopping the app pool using the IIS manager. Are you seeing a similar thing?

![image](https://cloud.githubusercontent.com/assets/6998306/15124753/c35a5e8e-1629-11e6-8c32-cc0e87a3f78e.png)

**Comment by moozzyk** [comment]
@PinpointTownes - this should not be happening. I guess it would be useful if you could provide @pan-wang with a dump.

**Comment by PinpointTownes** [comment]
Sure:

```
Nom de l’application défaillante w3wp.exe, version : 7.5.7601.17514, horodatage : 0x4ce7afa2
Nom du module défaillant : ntdll.dll, version : 6.1.7601.23392, horodatage : 0x56eb3625
Code d’exception : 0xc000070a
Décalage d’erreur : 0x0000000000060634
ID du processus défaillant : 0xb90
Heure de début de l’application défaillante : 0x01d1aa25a54b4a79
Chemin d’accès de l’application défaillante : c:\windows\system32\inetsrv\w3wp.exe
Chemin d’accès du module défaillant: C:\Windows\SYSTEM32\ntdll.dll
ID de rapport : f1e70cde-1618-11e6-b57c-c3e826fb1084
```

```
Récipient d’erreurs , type 0
Nom d’événement : APPCRASH
Réponse : Non disponible
ID de CAB : 0

Signature du problème : 
P1 : w3wp.exe
P2 : 7.5.7601.17514
P3 : 4ce7afa2
P4 : ntdll.dll
P5 : 6.1.7601.23392
P6 : 56eb3625
P7 : c000070a
P8 : 0000000000060634
P9 : 
P10 : 
```

**Comment by pan-wang** [comment]
This was fixed.

**Comment by Tratcher** [comment]
@pan-wang in which version?

**Comment by shirhatti** [comment]
Fixed in 7.1.1965.0
